src/content/docs/en/guides/markdown-content.mdx

```diff
@@ -9,223 +9,27 @@ import RecipeLinks from "~/components/RecipeLinks.astro";
-
-```mdx title="/src/pages/posts/post-1.mdx"
----
-layout: '../../layouts/BlogPostLayout.astro'
-title: 'My first MDX post'
----
-
-# {frontmatter.title}
-```
-
-### Using Components in MDX
-
-After installing the MDX integration, you can import and use both [Astro components](/en/basics/astro-components/#component-props) and [UI framework components](/en/guides/framework-components/#using-framework-components) in MDX (`.mdx`) files just as you would use them in any other Astro component.
-
-Don't forget to include a `client:directive` on your UI framework components, if necessary!
-
-See more examples of using import and export statements in the [MDX docs](https://mdxjs.com/docs/what-is-mdx/#esm).
-
-```mdx title="src/pages/about.mdx" {5-6} /<.+\/>/
----
-layout: ../layouts/BaseLayout.astro
-title: About me
----
-import Button from '../components/Button.astro';
-import ReactCounter from '../components/ReactCounter.jsx';
-
-I live on **Mars** but feel free to <Button title="Contact me" />.
-
-Here is my counter component, working in MDX:
-
-<ReactCounter client:load />
-```
-
-#### Assigning Custom Components to HTML elements
-
-With MDX, you can map Markdown syntax to custom components instead of their standard HTML elements. This allows you to write in standard Markdown syntax, but apply special component styling to selected elements.
-
-Import your custom component into your `.mdx` file, then export a `components` object that maps the standard HTML element to your custom component:
-
-```mdx title="src/pages/about.mdx"
-import Blockquote from '../components/Blockquote.astro';
-export const components = {blockquote: Blockquote}
-
-> This quote will be a custom Blockquote
-```
-
-
-```astro title="src/components/Blockquote.astro"
----
-const props = Astro.props;
----
-<blockquote {...props} class="bg-blue-50 p-4">
-  <span class="text-4xl text-blue-600 mb-2">“</span>
-  <slot /> <!-- Be sure to add a `<slot/>` for child content! -->
-</blockquote>
-```
-Visit the [MDX website](https://mdxjs.com/table-of-components/) for a full list of HTML elements that can be overwritten as custom components.
-
-## Importing Markdown
-
-You can import Markdown and MDX files directly into your Astro files. This gives you access to their Markdown content, as well as other properties such as frontmatter values that can be used within Astro's JSX-like expressions.
-
-You can import one specific page with an `import` statement, or multiple pages with [`Astro.glob()`](/en/guides/imports/#astroglob).
-
-```astro title="src/pages/index.astro"
----
-// Import a single file
-import * as myPost from '../pages/post/my-post.md';
-
-// Import multiple files with Astro.glob
-const posts = await Astro.glob('../pages/post/*.md');
----
-```
-
-When you import Markdown and MDX files in an Astro component, you get an object containing their [exported properties](#exported-properties).
-
-```md title="/src/pages/posts/great-post.md"
+```md title="/src/posts/great-post.md"
```

Comment:
It seems I can't add a suggestion here since some lines are not edited in between. So, I'm sorry if my suggestion is less readable. 😅 

I think there are some inconsistency between paths:
* `"/src/posts/great-post.md"` (l32)
* `import * as greatPost from '../pages/post/great-post.md';` (l43)
* `const posts = Object.values(await import.meta.glob('../post/*.md', { eager: true }));` (l44)

Should these paths be `src/posts/`, `src/post/` or `src/pages/post/` (`../pages` should be the same as `./` if I make no mistake)?

Also, if the example is there to illustrate both `import *` and `import.meta.glob`, I think we could rewrite it with a common use case (however, the glob can be a bit more complex for those not used to this syntax):
* an `index.md` file that contains specific metadata for the page that lists all the posts
* all posts in the same directory as `index.md`

In the following example, I assume that the posts (including `index.md`) are located in `/src/posts` and the page is `/src/pages/my-posts.astro`:

```astro
---
import * as postsIndex from '../posts/index.md';
const posts = Object.values(await import.meta.glob('../posts/!(index).md', { eager: true }));
---

<p>{postsIndex.frontmatter.title}</p>
<p>Written by: {postsIndex.frontmatter.author}</p>

<p>Posts Archive:</p>
<ul>
  {posts.map(post => <li><a href={post.url}>{post.frontmatter.title}</a></li>)}
</ul>
```
I haven't tested this example but it seems to me that Vite's `import.meta.glob` and Content Layer are both based on `fast-glob` implementation, so this should work.

Reply:
Thanks, you're right! I get what you mean and I believe I've updated for a (simple!) case that is now accurate.

I don't want to confuse people any more than necessary, so I'm making the paths consistent and not making people need to assume what the contents of that folder are. 😄